lib/ansible/modules/packaging/os/rhsm_repository.py

```diff
@@ -0,0 +1,260 @@
+                    }
+            repo_result.append(repo)
+    return repo_result
+
+
+def repository_list(module, list_parameter):
+    # Get RHSM repository list and format it for the user
+    repo = get_repository_list(module, list_parameter)
+    module.exit_json(changed=False, repositories=repo)
+
+
+def repository_modify(module, state, name):
+    name = set(name)
+    current_repo_list = get_repository_list(module, 'list')
+    updated_repo_list = deepcopy(current_repo_list)
+    matched_existing_repo = {}
+    for repoid in name:
+        matched_existing_repo[repoid] = []
+        for idx, repo in enumerate(current_repo_list):
+            if fnmatch(repo['id'], repoid):
+                matched_existing_repo[repoid].append(repo)
+                # Update current_repo_list to return it as result variable
+                updated_repo_list[idx]['enabled'] = True if state == 'enabled' else False
+
+    changed = False
+    results = []
+    diff_before = ""
+    diff_after = ""
+    rhsm_arguments = ['repos']
+
+    for repoid in matched_existing_repo:
+        if len(matched_existing_repo[repoid]) == 0:
+            results.append("%s is not a valid repository ID" % repoid)
+            module.fail_json(results=results, msg="%s is not a valid repository ID" % repoid)
+        for repo in matched_existing_repo[repoid]:
+            if state == 'disabled':
+                if repo['enabled']:
+                    changed = True
+                    diff_before += "Repository '%s' is enabled for this system\n" % repo['id']
+                    diff_after += "Repository '%s' is disabled for this system\n" % repo['id']
+                results.append("Repository '%s' is disabled for this system" % repo['id'])
+                rhsm_arguments += ['--disable', repo['id']]
+            elif state == 'enabled':
+                if not repo['enabled']:
+                    changed = True
+                    diff_before += "Repository '%s' is disabled for this system\n" % repo['id']
+                    diff_after += "Repository '%s' is enabled for this system\n" % repo['id']
+                results.append("Repository '%s' is enabled for this system" % repo['id'])
+                rhsm_arguments += ['--enable', repo['id']]
+
+    diff = {'before': diff_before,
+            'after': diff_after,
+            'before_header': "RHSM repositories",
+            'after_header': "RHSM repositories"}
+
+    if module.check_mode:
+        module.exit_json(results=results, changed=changed, repositories=updated_repo_list, diff=diff)
+
+    rc, out, err = run_subscription_manager(module, rhsm_arguments)
+    results = out.split('\n')
+    module.exit_json(results=results, changed=changed, repositories=updated_repo_list, diff=diff)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='list'),
+            state=dict(choices=['enabled', 'disabled', 'list', 'list_enabled', 'list_disabled'],
```

Comment:
Please format this like this:

```python
            state=dict(
                choices=['enabled', 'disabled', 'list', 'list_enabled', 'list_disabled'],
                default='enabled'),
```

Reply:
Changed in the last commit, thanks for the review